transit_odp/publish/tests/test_views.py

```diff
@@ -1633,6 +1633,146 @@ def test_require_attention_empty_search_box(
+            f"{self.form1_name}-licence": licence.id,
+            "submit": "submit",
+        }
+        step2_data = {
+            f"{self.wizard_name}-current_step": "provide_operating_dates",
+            f"{self.form2_name}-registration_code": self.REGISTRATION_CODE,
+            f"{self.form2_name}-start": "2023-01-05",
+            f"{self.form2_name}-end": "2023-01-15",
+            "submit": "submit",
+        }
+        publish_client.post(
+            self.url,
+            data=step1_data,
+        )
+        response = publish_client.post(self.url, data=step2_data)
+
+        assert (
+            response.context_data["form"].errors["registration_code"][0]
+            == "This service code has already been set up with a date range"
+        )
+
+    def test_seasonal_service_errors_on_end_before_start(self, publish_client):
+        licence = BODSLicenceFactory(organisation=self.org)
+        publish_client.force_login(user=self.user)
+        step1_data = {
+            f"{self.wizard_name}-current_step": "select_psv_licence_number",
+            f"{self.form1_name}-licence": licence.id,
+            "submit": "submit",
+        }
+        step2_data = {
+            f"{self.wizard_name}-current_step": "provide_operating_dates",
+            f"{self.form2_name}-registration_code": self.REGISTRATION_CODE,
+            f"{self.form2_name}-start": "2023-01-15",
+            f"{self.form2_name}-end": "2023-01-5",
+            "submit": "submit",
+        }
+        publish_client.post(
+            self.url,
+            data=step1_data,
+        )
+        response = publish_client.post(self.url, data=step2_data)
+
+        assert (
+            response.context_data["form"].errors["__all__"][0]
+            == "Start date must be earlier than end date"
+        )
+
+    @pytest.mark.parametrize(
+        "start,end", [("", "2023-01-15"), ("2023-01-05", ""), ("", "")]
+    )
+    def test_seasonal_service_errors_on_empty_dates(self, publish_client, start, end):
+        licence = BODSLicenceFactory(organisation=self.org)
+        publish_client.force_login(user=self.user)
+        step1_data = {
+            f"{self.wizard_name}-current_step": "select_psv_licence_number",
+            f"{self.form1_name}-licence": licence.id,
+            "submit": "submit",
+        }
+        step2_data = {
+            f"{self.wizard_name}-current_step": "provide_operating_dates",
+            f"{self.form2_name}-registration_code": self.REGISTRATION_CODE,
+            f"{self.form2_name}-start": start,
+            f"{self.form2_name}-end": end,
+            "submit": "submit",
+        }
+        publish_client.post(
+            self.url,
+            data=step1_data,
+        )
+        response = publish_client.post(self.url, data=step2_data)
+
+        errors = response.context_data["form"].errors
+        for error in errors.values():
+            assert error[0] == "This field is required."
```

Comment:
I thought the conclusion of the KPMG meeting was to change this to "This date is required" because they didn't like the word "field". Did I misunderstand?

Reply:
Scratch that, I can see you did it in a later commit.